### PR TITLE
pidstore: request next pid from legacy

### PR DESCRIPTION
**Pull request by jacenkow**
* Adds a method, which allows for reservation of new PID
  on legacy system.

* Adds test.

Signed-off-by: Grzegorz Jacenków <grzegorz.jacenkow@cern.ch>

### CODE REVIEW

File: inspirehep/config.py [hunk 1]
```diff
@@ -1149,6 +1149,10 @@ def _(x):
     }
 }
 
+# Legacy PID provider
+# ===================
+LEGACY_PID_PROVIDER = None # e.g. "http://server-name/batchuploader/allocaterecord"
```
Comment (kaplun) [inline inspirehep/config.py]: Please replace `example.org` instead of `server-name`, since this is the default recommended host to use in examples.

**Comment by jacenkow** [comment]
@kaplun it's 🍏

File: inspirehep/modules/pidstore/providers.py [hunk 2]
```diff
@@ -28,11 +28,28 @@
...
 
+from flask import current_app
+import requests
+
 from invenio_pidstore.models import PIDStatus, RecordIdentifier
 
 from invenio_pidstore.providers.base import BaseProvider
 
 
+def _get_next_pid_form_legacy():
```
Comment (jacquerie) [inline inspirehep/modules/pidstore/providers.py]: s/form/from/

File: inspirehep/modules/pidstore/providers.py [hunk 3]
```diff
@@ -53,7 +70,11 @@ def create(cls, object_type=None, object_uuid=None, **kwargs):
         """Create a new record identifier."""
         # Request next integer in recid sequence.
         if 'pid_value' not in kwargs:
-            kwargs['pid_value'] = str(RecordIdentifier.next())
+            # Check if the config specifies legacy pid provider.
+            if current_app.config.get('LEGACY_PID_PROVIDER'):
+                kwargs['pid_value'] = _get_next_pid_form_legacy()
```
Comment (jacquerie) [inline inspirehep/modules/pidstore/providers.py]: s/form/from/

File: inspirehep/modules/pidstore/providers.py [hunk 4]
```diff
@@ -28,11 +28,28 @@
 
 import os
 
+from flask import current_app
+import requests
```
Comment (jacquerie) [inline inspirehep/modules/pidstore/providers.py]: switch these two

File: inspirehep/modules/pidstore/providers.py (hunk 2 above)
Comment (jacenkow) [inline inspirehep/modules/pidstore/providers.py]: Good catch!

File: inspirehep/modules/pidstore/providers.py [hunk 5]
```diff
@@ -28,11 +28,28 @@
...
+
 from invenio_pidstore.models import PIDStatus, RecordIdentifier
 
 from invenio_pidstore.providers.base import BaseProvider
 
 
+def _get_next_pid_form_legacy():
+    """This method sends a request to the legacy system to reserve next
+    available pid and send the value back.
+    """
```
Comment (jacquerie) [inline inspirehep/modules/pidstore/providers.py]: Better:
```python
"""Reserve the next pid on legacy.

Sends a request to a legacy instance to reserve the next available
identifier, and returns it to the caller.
```

From [PEP257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings): 
> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description.

File: tests/integration/conftest.py [hunk 6]
```diff
@@ -95,3 +96,10 @@ def small_app(request):
         es.indices.refresh('records-hep')
 
         yield app
+
+
+@pytest.yield_fixture
+def httppretty_mock():
```
Comment (jacquerie) [inline tests/integration/conftest.py]: One less p

File: tests/integration/test_pidstore.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        'LEGACY_PID_PROVIDER': 'http://server/batchuploader/allocaterecord',
+    }
+
+    with app.app_context():
+        with mock.patch.dict(app.config, extra_config):
+            httpretty.register_uri(
+                httpretty.GET,
+                "http://server/batchuploader/allocaterecord",
+                content_type="application/json",
+                body=u'3141592',
```
Comment (jacquerie) [inline tests/integration/test_pidstore.py]: tau > pi : )

File: inspirehep/modules/pidstore/providers.py [hunk 8]
```diff
@@ -53,7 +70,11 @@ def create(cls, object_type=None, object_uuid=None, **kwargs):
         """Create a new record identifier."""
         # Request next integer in recid sequence.
         if 'pid_value' not in kwargs:
-            kwargs['pid_value'] = str(RecordIdentifier.next())
+            # Check if the config specifies legacy pid provider.
```
Comment (jacquerie) [inline inspirehep/modules/pidstore/providers.py]: No need for this comment, the code is already saying that.

File: tests/integration/test_pidstore.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+from __future__ import absolute_import, division, print_function
+
+import httpretty
+import mock
+
+from inspirehep.modules.pidstore.providers import InspireRecordIdProvider
+
+
+def test_getting_next_recid_from_legacy(httppretty_mock, app):
```
Comment (jacquerie) [inline tests/integration/test_pidstore.py]: One less p here as well.

**Comment by jacquerie** [comment]
Better commit message:
```
pidstore: request next pid from legacy

* Adds a method that reserves a new pid on legacy (closes #1356).

Signed-off-by: Grzegorz Jacenków <grzegorz.jacenkow@cern.ch>
```

**Comment by jacquerie** [comment]
:+1:, but some details were off.

File: tests/integration/test_pidstore.py (hunk 7 above)
Comment (kaplun) [inline tests/integration/test_pidstore.py]: I don't get it :)

Comment (jacquerie) [inline tests/integration/test_pidstore.py]: https://www.youtube.com/watch?v=jG7vhMMXagQ

**Pull request closed without merging**